### ISSUE TITLE
Ignore `__doc__` from parent classes

### ISSUE THREAD

**Issue by WillDaSilva** [mkdocstrings/griffe#55]
**Is your feature request related to a problem? Please describe.**
Currently Griffe will fall-back to extracting docstrings from the `__doc__` attribute if it cannot obtain them from the source file. When it does so, it will make use of the `__doc__` attribute of parent classes if the derived class does not have a docstring. This frequently leads to misleading documentation.

For example, if you create a class that inherits from `numbers.Real`, your class will have the following text picked up by Griffe unless you define a docstring:

```
>>> print(numbers.Real.__doc__)
To Complex, Real adds the operations that work on real numbers.

    In short, those are: a conversion to float, trunc(), divmod,
    %, <, <=, >, and >=.

    Real also provides defaults for the derived operations.
```

**Describe the solution you'd like**
Instead of using `getattr(x, '__doc__')`, Griffe can directly access the members of the class it is gathering docstrings from with `x.__dict__['__doc__']`.

This would make the behaviour more consistent with how Griffe extracts docstrings from regular Python modules, as for those it does not use the docstring from parent classes at all.

**Comment by WillDaSilva** [comment]
Special care would have to be taken to support classes that use `__slots__` instead of `__dict__`.

**Comment by pawamoy** [comment]
Agreed. Would you like to send a PR? The change needs to happen here:

https://github.com/mkdocstrings/griffe/blob/a9a378fc6e47678e08a22383879e4d01acd16b54/src/griffe/agents/inspector.py#L183

**Comment by WillDaSilva** [comment]
I'd be happy to. Thanks for pointing me in the right direction. I'll try to get it done this weekend.

**Comment by pawamoy** [comment]
Released in 0.14.0 :slightly_smiling_face:

**Comment by WillDaSilva** [comment]
Awesome! Wasn't expecting it to be released so soon. Thanks for the quick review.

**Comment by pawamoy** [comment]
For a long time I've been waiting before releases, to pack more things into each. But the result is that I simply forget about those features and bugfix to be released, and users wait for nothing. So, now, I try to release as soon as possible (still packing several things if I can). It's available faster to users, and it gets out of my mind :+1:
